contracts/ServiceAgreementRegistry.sol

```diff
@@ -269,50 +175,25 @@ contract ServiceAgreementRegistry is Initializable, OwnableUpgradeable, IService
         _establishServiceAgreement(newAgreementId, false);
     }
 
-    function clearEndedAgreement(address indexer, uint256 id) public {
-        require(id < indexerCsaLength[indexer], 'SA001');
+    function clearEndedAgreement(address indexer, uint256 id) external {
```

Comment:
`clearEndedAgreement` here is redundant with `removeEndedAgreement` in the new helper contract